src/SecTester.Scan/Scan.cs

```diff
@@ -96,6 +96,45 @@ public async Task Stop()
     yield return _scanState;
   }
 
+  public async Task Expect(Func<Scan, Task<bool>> predicate)
```

Comment:
We have already discussed that the user should be able to abort any long-running transient operation. To combine multiple sources, use `CancellationTokenSource.CreateLinkedTokenSource` 

```suggestion
  public async Task Expect(Func<Scan, Task<bool>> predicate,  CancellationToken? cancellationToken = default)
```